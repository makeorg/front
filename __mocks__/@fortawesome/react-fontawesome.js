
import React from 'react';

export const FontAwesomeIcon = ({ icon }) => (
  <svg
    lib="@fortawesome/react-fontawesome"
    icon={icon.iconName}
    prefix={icon.prefix}
  />
);
