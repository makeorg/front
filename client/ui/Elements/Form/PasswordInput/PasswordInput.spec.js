import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { faLock } from '@fortawesome/free-solid-svg-icons';
import { PasswordInput } from './index';

describe('PasswordInput', () => {
  const defaultProps = {
    type: 'foo',
    name: 'bar',
    icon: faLock,
    value: 'baz',
    label: 'qux',
    handleChange: () => {},
    togglePasswordIsDisplayed: () => {}
  };

  it('must match the snapshot with default Props', () => {
    const component = renderer.create(
      <PasswordInput {...defaultProps} />
    );
    expect(component).toMatchSnapshot();
  });

  it('must return the diff between snapshot with a negative vs a postive Tab Index', () => {
    const NegativeTabIndex = renderer.create(
      <PasswordInput {...defaultProps} tabIndex="-1" />
    );
    const PositiveTabIndex = renderer.create(
      <PasswordInput {...defaultProps} tabIndex="0" />
    );
    expect(snapshotDiff(NegativeTabIndex, PositiveTabIndex)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when password is displayed vs hidden', () => {
    const ShowPassword = renderer.create(
      <PasswordInput {...defaultProps} passwordIsDisplayed />
    );
    const HidePassword = renderer.create(
      <PasswordInput {...defaultProps} passwordIsDisplayed={false} />
    );
    expect(snapshotDiff(ShowPassword, HidePassword)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when password is required or optionnal', () => {
    const RequiredPassword = renderer.create(
      <PasswordInput {...defaultProps} required />
    );
    const OptionnalPassword = renderer.create(
      <PasswordInput {...defaultProps} required={false} />
    );
    expect(snapshotDiff(RequiredPassword, OptionnalPassword)).toMatchSnapshot();
  });
});
