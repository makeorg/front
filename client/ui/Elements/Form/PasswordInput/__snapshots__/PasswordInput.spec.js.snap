// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput must match the snapshot with default Props 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  padding: 0;
  border: none;
  background: transparent;
  background-color: transparent;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 0 1.125rem;
  border-radius: 1.875rem;
  border: 0.0625rem solid rgba(0,0,0,0.2);
  margin-bottom: 0.9375rem;
  background: rgb(242,242,242);
  background-color: rgb(242,242,242);
  border-color: rgba(0,0,0,0.2);
}

.c2 {
  border: none;
  background: transparent;
  background-color: transparent;
  padding: 0 0.625rem;
  width: 100%;
  color: rgba(0,0,0,0.55);
  font-size: 0.875rem;
  line-height: 2.188rem;
}

.c1 {
  width: 1.875rem;
  color: rgb(232,24,68);
  font-size: 1rem;
}

@media (min-width:47.94rem) {
  .c2 {
    font-size: 1rem;
    line-height: 2.5rem;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    font-size: 1.125rem;
  }
}

<div
  className="c0"
>
  <label
    aria-label="qux"
    className="c1"
    htmlFor="bar"
  >
    <svg
      icon="lock"
      lib="@fortawesome/react-fontawesome"
      prefix="fas"
    />
  </label>
  <input
    aria-required={false}
    className="c2"
    id="bar"
    name="bar"
    onChange={[Function]}
    placeholder="qux"
    required={false}
    tabIndex={0}
    type="foo"
    value="baz"
  />
  <button
    aria-hidden={true}
    className="c3"
    onClick={[Function]}
    tabIndex={0}
  >
    <svg
      icon="eye"
      lib="@fortawesome/react-fontawesome"
      prefix="fas"
    />
  </button>
</div>
`;

exports[`PasswordInput must return the diff between snapshot when password is displayed vs hidden 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -19,21 +19,21 @@
      name="bar"
      onChange={[Function handleChange]}
      placeholder="qux"
      required={false}
      tabIndex={0}
-     type="text"
+     type="foo"
      value="baz"
    />
    <button
      aria-hidden={true}
-     className="sc-1wm8iti-1 ntcd9w-9 fbihzC"
+     className="sc-1wm8iti-1 fncopd"
      onClick={[Function togglePasswordIsDisplayed]}
      tabIndex={0}
    >
      <svg
-       icon="eye-slash"
+       icon="eye"
        lib="@fortawesome/react-fontawesome"
        prefix="fas"
      />
    </button>
  </div>
`;

exports[`PasswordInput must return the diff between snapshot when password is required or optionnal 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -11,17 +11,17 @@
        lib="@fortawesome/react-fontawesome"
        prefix="fas"
      />
    </label>
    <input
-     aria-required={true}
+     aria-required={false}
      className="ntcd9w-0 ntcd9w-7 osRhJ"
      id="bar"
      name="bar"
      onChange={[Function handleChange]}
      placeholder="qux"
-     required={true}
+     required={false}
      tabIndex={0}
      type="foo"
      value="baz"
    />
    <button
`;

exports[`PasswordInput must return the diff between snapshot with a negative vs a postive Tab Index 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -18,19 +18,19 @@
      id="bar"
      name="bar"
      onChange={[Function handleChange]}
      placeholder="qux"
      required={false}
-     tabIndex="-1"
+     tabIndex="0"
      type="foo"
      value="baz"
    />
    <button
      aria-hidden={true}
      className="sc-1wm8iti-1 fncopd"
      onClick={[Function togglePasswordIsDisplayed]}
-     tabIndex="-1"
+     tabIndex="0"
    >
      <svg
        icon="eye"
        lib="@fortawesome/react-fontawesome"
        prefix="fas"
`;
